rdmo/projects/answers.py

```diff
@@ -0,0 +1,252 @@
+from collections import defaultdict
+
+from rdmo.core.utils import markdown2html
+from rdmo.questions.models import Catalog, Page, Question, QuestionSet, Section
+
+
+class AnswerTree:
+
+    def __init__(self, catalog, values):
+        self.catalog = catalog
+        self.values = values
+
+        self.sets = values.compute_sets()
+        self.conditions = catalog.conditions.in_bulk()
+
+        # buffer for the resolved conditions: self.resolved_conditions[element][parent_set]
+        self.resolved_conditions = defaultdict(lambda: defaultdict(dict))
+
+    def compute(self):
+        # Main function of this class, which Computes the answer tree recursively.
+        # First, it computes the catalog, section, and page nodes.
+        # Then, it alternates between (value)set and questionset nodes until it reaches
+        # the question nodes, which include the corresponding values as well as how much
+        # this question counts to the count and total values for the progress.
+        return self.compute_element_node(self.catalog)
+
+    def compute_element_node(self, element, parent_set=None):
+        # recursive function, which will be called for each element
+        element_node = {
+            'id': element.id,
+            'uri': element.uri,
+            'model': str(element._meta),
```

Comment:
could be changed to explicitly use the `_meta.label_lower` property, or kept as-is.
* https://github.com/django/django/blob/1b50da7e941d015c2eb9e36bfd2913e37726ac1d/django/db/models/options.py#L389